test/product-import.spec.coffee

```diff
@@ -470,3 +470,137 @@ describe 'ProductImport', ->
       updatedProduct = @import._ensureDefaults(sampleProduct)
       expect(updatedProduct).toEqual expectedProduct
 
+  describe ':: Custom reference resolution', ->
+
+    it ' :: should detect reference type attribute', ->
+      sampleReferenceAttribute =
+        name: 'foobar'
+        value:
+          value: 'some value'
+          resolvePredicate: 'some predicate query'
+          endpoint: 'some endpoint'
```

Comment:
Also, this custom logic should be documented in the source code, otherwise it's hard to understand the reason